File: excel-sheet-column-title/excel-sheet-column-title.py
def convertToTitle(columnNumber: int) -> str:
    result = ""
    totalLetters = 26
    asciiInitCode = 65

    while columnNumber > 0:
        columnNumber -= 1
        charCode = (columnNumber % totalLetters) + asciiInitCode
        result = chr(charCode) + result
        columnNumber //= totalLetters

    return result

print(convertToTitle(1))    # "A"
print(convertToTitle(28))   # "AB"
print(convertToTitle(701))  # "ZY"

File: excel-sheet-column-title/excel-sheet-column-title.go
package main

import (
	"fmt"
)

func convertToTitle(columnNumber int) string {
	result := ""
	totalLetters := 26
	asciiInitCode := 65

	for columnNumber > 0 {
		columnNumber--
		charCode := (columnNumber % totalLetters) + asciiInitCode
		result = string(rune(charCode)) + result
		columnNumber /= totalLetters
	}

	return result
}

func main() {
	fmt.Println(convertToTitle(1))   // "A"
	fmt.Println(convertToTitle(28))  // "AB"
	fmt.Println(convertToTitle(701)) // "ZY"
}
